### ISSUE TITLE
Disk provisioning settings are not correctly applied based on provided arguments

### ISSUE THREAD

**Issue by nathancarmona** [hashicorp/terraform-provider-vsphere#1303]
### Terraform Version

`0.13.4 `

### vSphere Provider Version

`1.24.3.`

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
resource "vsphere_virtual_machine" "main" {
  name             = var.name
  resource_pool_id = data.vsphere_compute_cluster.main.resource_pool_id
  guest_id         = data.vsphere_virtual_machine.template.guest_id
  network_interface {
    network_id   = data.vsphere_network.main.id
    adapter_type = data.vsphere_virtual_machine.template.network_interface_types[0]
  }

  num_cpus = var.resources_cpu
  memory   = var.resources_mem

  scsi_type = data.vsphere_virtual_machine.template.scsi_type

  disk {
    # other disk thats thin provisioned and doesn't have issues
  }
  disk {
    label            = var.label
    unit_number      = var.unit_number
    size             = var.size
    datastore_id     = data.vsphere_datastore.main.id
    disk_mode        = var.disk_mode
    eagerly_scrub    = var.eagerly_scrub # true
    thin_provisioned = var.thin_provisioned # false
    io_limit         = var.io_limit
    io_reservation   = var.io_reservation
    io_share_level   = var.io_share_level
    io_share_count   = var.io_share_count
    }
  }

  clone {
    template_uuid = data.vsphere_virtual_machine.template.id
  }
}
```

### Debug Output

Apply:
https://gist.github.com/nathancarmona/9bd173cb6bc2f1a94bbe55ae5da964c3

Plan:
https://gist.github.com/nathancarmona/5ab9f0b5cb0af970bdeada1aee1bfb4f

### Plan Output

``` bash
 disk {
          + attach            = false
          + controller_type   = "scsi"
          + datastore_id      = "datastore-11"
          + device_address    = (known after apply)
          + disk_mode         = "persistent"
          + disk_sharing      = "sharingNone"
          + eagerly_scrub     = true
          + io_limit          = -1
          + io_reservation    = 0
          + io_share_count    = 0
          + io_share_level    = "normal"
          + keep_on_remove    = false
          + key               = 0
          + label             = "disk1"
          + path              = (known after apply)
          + size              = 81
          + storage_policy_id = (known after apply)
          + thin_provisioned  = false
          + unit_number       = 1
          + uuid              = (known after apply)
          + write_through     = false
        }
```

### Expected Behavior

Disk should have been provisioned thick with eager zero.

### Actual Behavior

Disk was provisioned thick with lazy zero. Running a plan or apply again fails with:

``` bash
Error: disk.1: virtual disk "disk1": cannot change the value of "eagerly_scrub" - (old: false new: true)

  on .terraform/modules/test_virtual_machine/virtual-machine.tf line 1, in resource "vsphere_virtual_machine" "main":
   1: resource "vsphere_virtual_machine" "main" {
```

### Steps to Reproduce

It does not seem to matter whether the disk in the template is set to eager or lazy zero, so a template set to either and Terraform resource set to `eager` still seems to result in a lazy zero'd disk. Manually creating a VM from the template in either configuration will produce an eager zero'd disk.

1. Create a VM template with a thick provisioned, eager zero'd disk
2. Apply a template with config the matches the template (thick, eager)
3. Note the vm that was created has a lazy zero'd disk
4. Run another plan/apply and note the error that the disk can't be changed from lazy to eager

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by tenthirtyam** [comment]
Adding label as this issue has been **confirmed.**

Ryan Johnson
Staff II Solutions Architect | VMware, Inc.

**Comment by tenthirtyam** [comment]
Updating the issue to reflect the higher-level issue as seen in duplicate #1358.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by tenthirtyam** [comment]
TL;DR: The disk provisioning settings are not being applied based on provided arguments given to `eagerly_scrubbed` and `thin_provisioned` for a clone.

When cloning a virtual machine, the virtual machine only retains the settings in which was original provisioned.

- thin >> thin
- lazy zero >> lazy zero
- eager zero >> eager zero

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by tenthirtyam** [comment]
Marking as closed in favor of the prior #1028.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.1028.
